src/finalfusion/compat/fasttext.py

```diff
@@ -0,0 +1,340 @@
+        storage = _read_ft_storage(inf, vocab)
+        norms = _normalize_ndarray_storage(storage[:len(vocab)])
+    return Embeddings(storage=storage,
+                      vocab=vocab,
+                      norms=norms,
+                      metadata=metadata)
+
+
+def write_fasttext(file: Union[str, bytes, int, PathLike], embeds: Embeddings):
+    """
+    Write embeddings in fastText format.
+
+    fastText requires Metadata with all expected keys for fastText configs:
+        * dims: int (inferred from model)
+        * window_size: int (default -1)
+        * min_count: int (default -1)
+        * ns: int (default -1)
+        * word_ngrams: int (default 1)
+        * loss: one of ``['HierarchicalSoftmax', 'NegativeSampling', 'Softmax']`` (default Softmax)
+        * model: one of ``['CBOW', 'SkipGram', 'Supervised']`` (default SkipGram)
+        * buckets: int (inferred from model)
+        * min_n: int (inferred from model)
+        * max_n: int (inferred from model)
+        * lr_update_rate: int (default -1)
+        * sampling_threshold: float (default -1)
+
+    ``dims``, ``buckets``, ``min_n`` and ``max_n`` are inferred from the model. If other values
+    are unspecified, a default value of ``-1`` is used for all numerical fields. Loss defaults
+    to ``Softmax``, model to ``SkipGram``. Unknown values for ``loss`` and ``model`` are
+    overwritten with defaults since the models are incompatible with fastText otherwise.
+
+    Some information from original fastText models gets lost e.g.:
+        * word frequencies
+        * n_tokens
+
+    Embeddings are un-normalized before serialization: if norms are present, each embedding is
+    scaled by the associated norm. Additionally, the original state of the embedding matrix is
+    restored, precomputation and l2-normalization of word embeddings is undone.
+
+    Only embeddings with a FastTextVocab or SimpleVocab can be serialized to this format.
+
+    Parameters
+    ----------
+    file : str, bytes, int, PathLike
+        Output file
+    embeds : Embeddings
+        Embeddings to write
+    """
+    with open(file, 'wb') as outf:
+        if not isinstance(embeds.vocab, (FastTextVocab, SimpleVocab)):
+            raise ValueError(
+                f'Expected FastTextVocab or SimpleVocab, not: {type(embeds.vocab).__name__}'
+            )
+        _write_binary(outf, "<ii", 793_712_314, 12)
+        _write_ft_cfg(outf, embeds)
+        _write_ft_vocab(outf, embeds.vocab)
+        _write_binary(outf, "<?QQ", 0, *embeds.storage.shape)
+        if isinstance(embeds.vocab, SimpleVocab):
+            _write_ft_storage_simple(outf, embeds)
+        else:
+            _write_ft_storage_subwords(outf, embeds)
+        _serialize_array_as_le(outf, embeds.storage)
+
+
+def _read_ft_header(file: BinaryIO):
+    """
+    Helper method to verify version and magic.
+    """
+    magic, version = _read_required_binary(file, "<ii")
+    if magic != 793_712_314:
```

Comment:
Named constant

Reply:
Defining file magic not as a magic number, the irony! Fixed tho ;)